server/honeybadger/plugins.py

```diff
@@ -26,18 +28,51 @@ def get_coords_from_google(aps):
         data['lng'] = jsondata['location']['lng']
     return data
 
-def get_coords_from_uniapple(ip):
-    logger.info('Geolocating via Uniapple API.')
-    url = 'http://uniapple.net/geoip/?ip={}'.format(ip)
-    content = urllib2.urlopen(url).read()
-    logger.info('Uniapple API response:\n{}'.format(content))
+def get_external_ip(ip):
+    logger.info('Resolving external IP via Ipify API.')
+    url = "https://api.ipify.org"   # IPs are often reported as internal when testing,
+                                    #   and can't be geolocated by ipstack or ipinfo
+                                    #   as a result. This API call remedies that.
+    content = urlopen(url).read()
+    return content.decode()
+
+def get_coords_from_ipstack(ip):
+    logger.info('Geolocating via Ipstack API.')
+    url = 'http://api.ipstack.com/{0}?access_key={1}'.format(get_external_ip(ip), app.config['IPSTACK_API_KEY'])
+    content = urlopen(url).read()
+    logger.info('Ipstack API response:\n{}'.format(content))
     jsondata = None
     try:
         jsondata = json.loads(content)
     except ValueError as e:
         logger.error('{}.'.format(e))
+
+    # Avoid the KeyError. For some reason, a successful API call to Ipstack doesn't include
+    #   the 'success' key in the json result, but a failed call does, and the value is False
+    if 'success' in jsondata and not jsondata['success']:
+        logger.info('Ipstack API call failed: {}'.format(jsondata['error']['type']))
+        # Return with no data so the caller knows to default to the fallback API
+        return None
+
     data = {'lat':None, 'lng':None}
     if jsondata:
         data['lat'] = jsondata['latitude']
         data['lng'] = jsondata['longitude']
     return data
+
+def get_coords_from_ipinfo(ip):
+    # New fallback, ipinfo doesn't require an API key for a certain number of API calls
+    logger.info('Geolocating via Ipinfo.io API.')
+    url = 'https://ipinfo.io/{}'.format(get_external_ip(ip))
+    content = urlopen(url).read()
```

Comment:
If you're ok with making the requests module a dependency (I am) we could use that instead of all these calls to urlopen and manually loading the JSON.

Reply:
I am more familiar with the requests module, so I would be fine with refactoring that in. I only used the urllib functions because that was already used and I didn't want to rock the boat too much on the first PR.

Reply:
Please go ahead and integrate the requests module in this PR. Add it to the requirements file and replace urllib with requests. I think that's a good change.

Reply:
If I make changes to this branch, will they automatically come through on this PR?

Reply:
I believe so. Any new commits to the branch should get pulled into the request. I've actually never been on that side of things, but i know people do that all the time.

Reply:
Alright. I'll give it a shot.